File: 102_Binary Tree Level Order Traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    if root == nil {
        return [][] int {}
    }
    
    result := [][] int {}
    
    bfs(&result, 0, root)
    
    return result
}

func bfs(result *[][] int, level int, node *TreeNode) {
    if node == nil {
        return
    }
    
    if len(*result) == level {
        *result = append(*result, [] int {})
    }
    
    (*result)[level] = append((*result)[level], node.Val)
    
    for _, val := range []*TreeNode{node.Left, node.Right} {
        bfs(result, level + 1, val)
    }
}

File: 102_Binary Tree Level Order Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if(!root) {
        return [];
    }
    
    const result = [];
    const stack = [];
    stack.push(root);
    
    while(stack.length > 0) {
        const size = stack.length;
        const temp = []
        
        for(let i = 0; i < size; ++i) {
            const node = stack.shift();
            temp.push(node.val);
            if(node.left) {
                stack.push(node.left)
            }
            
            if(node.right) {
                stack.push(node.right);
            }
        }
        result.push(temp);
    }
    return result;
};